mindsdb/integrations/handlers/whatsapp_handler/whatsapp_handler.py

```diff
@@ -0,0 +1,406 @@
+
+            if arg1 == 'sent_at' and arg2 is not None:
+                date = parse_local_date(arg2)
+                if op == '>':
+                    params['date_sent_after'] = date
+                elif op == '<':
+                    params['date_sent_before'] = date
+                else:
+                    raise NotImplementedError
+
+                # also add to post query filter because date_sent_after=date1 will include date1
+                filters.append([op, arg1, arg2])
+
+            elif arg1 == 'sid':
+                if op == '=':
+                    params['sid'] = arg2
+                else:
+                    NotImplementedError('Only  "from_number=" is implemented')
+
+            elif arg1 == 'from_number':
+                if op == '=':
+                    params['from_number'] = arg2
+                else:
+                    NotImplementedError('Only  "from_number=" is implemented')
+
+            elif arg1 == 'to_number':
+                if op == '=':
+                    params['to_number'] = arg2
+                else:
+                    NotImplementedError('Only  "to_number=" is implemented')
+
+            else:
+                filters.append([op, arg1, arg2])
+
+        # Fetch messages based on the filters
+        result = self.handler.fetch_messages(params, df=True)
+
+        # filter targets
+        result = filter_dataframe(result, filters)
+
+        # If limit is specified
+        if query.limit is not None:
+            result = result[:int(query.limit.value)]
+
+        # project targets
+        result = project_dataframe(result, query.targets, self.get_columns())
+
+        return result
+
+    def get_columns(self):
+        return [
+            'sid',
+            'from_number',
+            'to_number',
+            'body',
+            'direction',
+            'msg_status',
+            'sent_at',  # datetime.strptime(str(msg.date_sent), '%Y-%m-%d %H:%M:%S%z'),
+            'account_sid',
+            'price',
+            'price_unit',
+            'api_version',
+            'uri'
+        ]
+
+    def insert(self, query: ast.Insert):
+        """
+        Sends a whatsapp message
+
+        Args:
+            body: message body
+            from_number: number from which to send the message
+            to_number: number to which message will be sent
+        """
+
+        # get column names and values from the query
+        columns = [col.name for col in query.columns]
+
+        ret = []
+
+        insert_params = ["body", "from_number", "to_number"]
+        for row in query.values:
+            params = dict(zip(columns, row))
+
+            # Check text length
+            max_text_len = 1500
+            text = params["body"]
+            words = re.split('( )', text)
+            messages = []
+
+            # Check text pattern
+            text2 = ''
+            pattern = r'http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+'
```

Comment:
Do we need to replace urls? If so can we better comment this so we know what happens here

Reply:
I updated the comments.